### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by acolombo1**
- Add unit tests to classes:
  - Book, Person, Student, Teacher, Classroom, Rental
  - Classes for the Decorator design pattern

### CODE REVIEW

File: spec/trim_decorator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  before :each do
+    @teacher = Teacher.new(55, 'Science', 'James Smith Jr.')
+  end
+  context 'test trimmer' do
+    it 'should return a name trimmed' do
+      @trimmed = TrimmerDecorator.new(@teacher)
+      expect(@trimmed.correct_name).to eql 'James Smit'
+    end
+  end
+end
```
Comment (Whoistolu) [inline spec/trim_decorator_spec.rb]: - [ ] [REQUIRED] Kindly ensure that the test for the `TrimmerDecorator` class passes as it currently throws an error that does not identify the teacher constant.
Screenshot from 2022-07-28 00-17-21

File: spec/cap_decorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    @classroom = Classroom.new('History')
+    @student = Student.new(55, @classroom, 'alexis')
+  end
+  context 'test capitalizer' do
+    it 'should return a name capitalized' do
+      @capitalized = CapitalizeDecorator.new(@student)
+      expect(@capitalized.correct_name).to eql 'Alexis'
+    end
+  end
+end
```
Comment (Whoistolu) [inline spec/cap_decorator_spec.rb]: - [ ] [REQUIRED] Kindly ensure that the test for the `CapitalizeDecorator` class passes as it currently throws an error that does not identify the classroom constant.
Screenshot from 2022-07-28 00-21-58

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi Team
@acolombo1 and @alexpuente0 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/trim_decorator_spec.rb (hunk 1 above)
Comment (acolombo1) [reply spec/trim_decorator_spec.rb]: When running this test alone, it yields the error because of missing requires. I'll fix it

**Review by rustam-tolipov — APPROVED** [review]
## Hello @acolombo1 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**